### PR TITLE
AG-12966 Pyramid series

### PR DESCRIPTION
**Pull request by jacobp100**

### CODE REVIEW

File: packages/ag-charts-enterprise/src/series/pyramid/pyramidSeries.ts [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+        }
+
+        const nodeData: PyramidNodeDatum[] = [];
+        const labelData: PyramidNodeLabelDatum[] = [];
+        let yStart = 0;
+        processedData.data.forEach(({ datum, values }, index) => {
+            const xValue: string = values[xIdx];
+            const yValue = Number(values[yIdx]);
+
+            let yEnd = yStart + yValue;
```
Comment (jacobp100) [inline packages/ag-charts-enterprise/src/series/pyramid/pyramidSeries.ts]:
```suggestion
            const yEnd = yStart + yValue;
```

File: packages/ag-charts-community/src/chart/series/seriesAreaFocusManager.ts
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/series/seriesAreaFocusManager.ts:1]: Looks like a bad merge, this file got removed earlier?

File: packages/ag-charts-enterprise/src/series/pyramid/pyramidSeries.ts [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+        await this.updateLabelNodes({ labelSelection });
+
+        this.stageLabelSelection = await this.updateStageLabelSelection({ stageLabelData, stageLabelSelection });
+        await this.updateStageLabelNodes({ stageLabelSelection });
+
+        this.highlightDatumSelection = await this.updateDatumSelection({
+            nodeData: highlightedDatum != null ? [highlightedDatum] : [],
+            datumSelection: highlightDatumSelection,
+        });
+        await this.updateDatumNodes({ datumSelection: highlightDatumSelection, isHighlight: true });
```
Comment (alantreadway) [inline packages/ag-charts-enterprise/src/series/pyramid/pyramidSeries.ts:291]: I feel like logically these calls should live in `updateSelections()`?

Comment (alantreadway) [reply packages/ag-charts-enterprise/src/series/pyramid/pyramidSeries.ts:291]: Discussed offline, let's follow-up with a general refactor?

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**